js/calendar.js

```diff
@@ -530,19 +534,36 @@ if(!String.prototype.formatNum) {
 				e.end_hour = f.getDate() + ' ' + $self.locale['ms' + f.getMonth()] + ' ' + e.end_hour;
 			}
 
-			if(e.start < start.getTime() && e.end > end.getTime()) {
-				data.all_day.push(e);
-				return;
-			}
+			if(!show_fits_time_events) {
```

Comment:
I think we can use var `$self.options.show_events_which_fits_time`  right here and get rid of `var show_fits_time_events = $self.options.show_events_which_fits_time;`